guides/source/active_record_migrations.md

```diff
@@ -1307,3 +1623,118 @@ special comment like this:
+      def change
+        enable_extension "uuid-ossp"
+      end
+    end
+    ```
+
+2. Rails Configuration
+
+    In your Rails application configuration file (`config/application.rb`), add
+    the following line to configure Rails to generate UUIDs as primary keys by
+    default:
+
+    ```ruby
+    config.generators do |g|
+      g.orm :active_record, primary_key_type: :uuid
+    end
+    ```
+
+    This setting instructs Rails to use UUIDs as the default primary key type
+    for ActiveRecord models.
+
+3. Adding References with UUIDs:
+
+    When creating associations between models using references, ensure that you
+    specify the data type as :uuid to maintain consistency with the primary key
+    type. For example:
+
+    ``` ruby
+    create_table :posts, id: :uuid do |t|
+      t.references :author, type: :uuid, foreign_key: true
+      # Other columns...
+      t.timestamps
+    end
+    ```
+
+    In this example, the `author_id` column in the posts table references the
+    `id` column of the authors table. By explicitly setting the type to `:uuid`,
+    you ensure that the foreign key column matches the data type of the primary
+    key it references. Adjust the syntax accordingly for other associations and
+    databases.
+
+4. Migration Changes
+
+    When generating migrations for your models, you'll notice that it secifies
+    the id to be of type uuid:
+
+    ```bash
+      $ bin/rails g migration CreateAuthors
+    ```
+
+    ```ruby
+    class CreateAuthors < ActiveRecord::Migration[8.0]
+      def change
+        create_table :authors, id: :uuid do |t|
+          t.timestamps
+        end
+      end
+    end
+    ```
+
+    which results in the following schema:
+
+    ```ruby
+      create_table "authors", id: :uuid, default: -> { "gen_random_uuid()" }, force: :cascade do |t|
+      t.datetime "created_at", precision: 6, null: false
+      t.datetime "updated_at", precision: 6, null: false
+    end
+    ```
+
+    In this migration, the `id` column is defined as a UUID primary key with a
+    default value generated by the `gen_random_uuid()` function provided by the
+    `uuid-ossp` extension.
+
+UUIDs are guaranteed to be globally unique across different systems, making them
+suitable for distributed architectures. They also simplify integration with
+external systems or APIs by providing a unique identifier that doesn't rely on
+centralized ID generation, and unlike auto-incrementing integers, UUIDs don't
+expose information about the total number of records in a table, which can be
+beneficial for security purposes.
+
+However, UUIDs are also larger than integers which can lead to increased storage
+requirements, especially in large databases. They can also impact performance
+due to their size. Therefore, it's essential to evaluate the trade-offs and
+consider the specific requirements of your application before deciding to use
+UUIDs as primary keys.
```

Comment:
Perhaps it would be useful to also add disclaimer about how using UUID (and especially fully random versions) may affect BTREE indexing performance due to rebalancing of the tree?

Reply:
I agree. I discourage their use with my clients however some already have them in place, and the negative impacts to performance for v4 UUIDs is significant.

Fortunately for v7 UUIDs this will  go away eventually, but it's not yet standard.

I can offer a write-up on this if it would help.

I do recommend at a minimum suggesting UUIDs will definitely have worse performance for writes and reads compared with integer primary key and foreign keys, primarily by preventing the use of Btree indexes that would have otherwise been used had integer types been chosen instead.

Reply:
> Fortunately for v7 UUIDs this will go away eventually, but it's not yet standard.

[RFC 9562](https://datatracker.ietf.org/doc/html/rfc9562) was published on [May 7](https://datatracker.ietf.org/doc/rfc9562/history/) so that's not a problem anymore 😄

Reply:
Hmm, BTREE indexing performance specifics is probably out of scope for this guide.

The line about impacting "performance due to their size" seems like a repetition of the previous line (" UUIDs are also larger than integers which can lead to increased storage requirements, especially in large databases")

Maybe make it more generic and move the last comment to a NOTE?
```
However, UUIDs can also impact performance due to their size and are
harder to index. UUIDs will have worse performance for writes and reads
compared with integer primary keys and foreign keys. 

NOTE: Therefore, it's essential to evaluate the trade-offs and
consider the specific requirements of your application before deciding to use
UUIDs as primary keys.
```

Or maybe 
```
However, UUIDs are also larger than integers which can lead to increased storage
requirements, especially in large databases. Furthermore, UUID values don’t index
well due to their size and randomness.

NOTE: Therefore, it's essential to evaluate the trade-offs and
consider the specific requirements of your application before deciding to use
UUIDs as primary keys.
```